sdk-tests/src/test/java/io/dapr/it/methodinvoke/grpc/MethodInvokeIT.java

```diff
@@ -0,0 +1,145 @@
+                System.out.println("Invoke method messages : " + message);
+            }
+
+            Map<Integer, String> messages = client.invokeMethod(
+                daprRun.getAppName(),
+                "getMessages",
+                GetMessagesRequest.newBuilder().build(),
+                HttpExtension.POST, GetMessagesResponse.class).block().getMessagesMap();
+            assertEquals(10, messages.size());
+
+            // Delete one message.
+            client.invokeMethod(
+                daprRun.getAppName(),
+                "deleteMessage",
+                DeleteMessageRequest.newBuilder().setId(1).build(),
+                HttpExtension.POST).block();
+            messages = client.invokeMethod(
+                daprRun.getAppName(),
+                "getMessages",
+                GetMessagesRequest.newBuilder().build(),
+                HttpExtension.POST, GetMessagesResponse.class).block().getMessagesMap();
+            assertEquals(9, messages.size());
+
+            // Now update one message.
+            client.invokeMethod(
+                daprRun.getAppName(),
+                "postMessage",
+                PostMessageRequest.newBuilder().setId(2).setMessage("updated message").build(),
+                HttpExtension.POST).block();
+            messages = client.invokeMethod(
+                daprRun.getAppName(),
+                "getMessages",
+                GetMessagesRequest.newBuilder().build(),
+                HttpExtension.POST, GetMessagesResponse.class).block().getMessagesMap();
+            assertEquals("updated message", messages.get(2));
+        }
+    }
+
+    @Test
+    public void testInvokeTimeout() throws Exception {
+        try (DaprClient client = new DaprClientBuilder().build()) {
+            long started = System.currentTimeMillis();
+            SleepRequest req = SleepRequest.newBuilder().setSeconds(1).build();
+            String message = assertThrows(IllegalStateException.class, () ->
+                client.invokeMethod(daprRun.getAppName(), "sleep", req.toByteArray(), HttpExtension.POST)
+                    .block(Duration.ofMillis(10))).getMessage();
+            long delay = System.currentTimeMillis() - started;
+            assertTrue(delay <= 500);  // 500 ms is a reasonable delay if the request timed out.
+            assertEquals("Timeout on blocking read for 10 MILLISECONDS", message);
+        }
+    }
+
+    @Test
+    public void testInvokeException() throws Exception {
+        try (DaprClient client = new DaprClientBuilder().build()) {
+            SleepRequest req = SleepRequest.newBuilder().setSeconds(-9).build();
+            DaprException exception = assertThrows(DaprException.class, () ->
+                client.invokeMethod(daprRun.getAppName(), "sleep", req.toByteArray(), HttpExtension.POST).block());
+
+            assertEquals("UNKNOWN", exception.getErrorCode());
+            assertEquals("UNKNOWN: HTTP status code: 500", exception.getMessage());
```

Comment:
is this because the method invocation is always `http` ? 
if it is gRPC why is the error message saying HTTP.

Reply:
Let me check. It should not be since I am testing with gRPC client too.

Reply:
Fixed. Our ITs where not switching to gRPC.